File: src/main/java/com/wgzhao/addax/admin/model/SysItem.java
package com.wgzhao.addax.admin.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.Getter;
import lombok.Setter;

@Table(name="sys_item")
@Getter
@Setter
@Data
@Entity
@IdClass(SysItemPK.class)
public class SysItem {
    @Id
    @Column(name="dict_code")
    private short dictCode;

    @Id
    @Column(name="item_key", length = 255)
    private String itemKey;

    @Column(name="item_value", length = 2000)
    private String itemValue;

    @Column(name="remark", length = 4000)
    private String remark;
}

File: src/main/kotlin/com/wgzhao/addax/admin/model/SysItem.kt
package com.wgzhao.addax.admin.model

import io.swagger.v3.oas.annotations.media.Schema
import jakarta.persistence.*

@Table(name = "sys_item")
@Entity
@IdClass(SysItemPK::class)
data class SysItem(
    @Id
    @Column(name = "dict_code")
    @Schema(description = "字典编码，外键", example = "1000")
    var dictCode: Int,

    @Id
    @Column(name = "item_key", length = 255)
    @Schema(description = "字典项键", example = "SWITCH_TIME")
    var itemKey: String,

    @Column(name = "item_value", length = 2000)
    @Schema(description = "字典项值", example = "16:30")
    var itemValue: String,

    @Column(name = "remark", length = 4000)
    @Schema(description = "备注", example = "切日时间")
    var remark: String? = null
)
